From 0fc2e70b9160580bd55e2529fba3659512878cb4 Mon Sep 17 00:00:00 2001
From: Shaoyun Liu <Shaoyun.Liu@amd.com>
Date: Tue, 22 May 2018 11:45:41 -0400
Subject: [PATCH 4515/5725] drm/amdgpu: Update GFX info structure to match what
 vega20 used

Update to the latest version from the vbios team.

Signed-off-by: Shaoyun Liu <Shaoyun.Liu@amd.com>
Acked-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_atomfirmware.c | 8 +++-----
 drivers/gpu/drm/amd/include/atomfirmware.h       | 3 ++-
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_atomfirmware.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_atomfirmware.c
index 7014d58..2369158 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_atomfirmware.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_atomfirmware.c
@@ -354,11 +354,9 @@ int amdgpu_atomfirmware_get_gfx_info(struct amdgpu_device *adev)
 				le16_to_cpu(gfx_info->v24.gc_gsprim_buff_depth);
 			adev->gfx.config.double_offchip_lds_buf =
 				gfx_info->v24.gc_double_offchip_lds_buffer;
-			adev->gfx.cu_info.wave_front_size = gfx_info->v24.gc_wave_size;
-			adev->gfx.cu_info.max_waves_per_simd =
-				le16_to_cpu(gfx_info->v24.gc_max_waves_per_simd);
-			adev->gfx.cu_info.max_scratch_slots_per_cu =
-				gfx_info->v24.gc_max_scratch_slots_per_cu;
+			adev->gfx.cu_info.wave_front_size = le16_to_cpu(gfx_info->v24.gc_wave_size);
+			adev->gfx.cu_info.max_waves_per_simd = le16_to_cpu(gfx_info->v24.gc_max_waves_per_simd);
+			adev->gfx.cu_info.max_scratch_slots_per_cu = gfx_info->v24.gc_max_scratch_slots_per_cu;
 			adev->gfx.cu_info.lds_size = le16_to_cpu(gfx_info->v24.gc_lds_size);
 			return 0;
 		default:
diff --git a/drivers/gpu/drm/amd/include/atomfirmware.h b/drivers/gpu/drm/amd/include/atomfirmware.h
index fd5e80c..c6c1666 100644
--- a/drivers/gpu/drm/amd/include/atomfirmware.h
+++ b/drivers/gpu/drm/amd/include/atomfirmware.h
@@ -1240,7 +1240,6 @@ struct  atom_gfx_info_v2_4 {
   uint8_t active_cu_per_sh;
   uint8_t active_rb_per_se;
   uint16_t gcgoldenoffset;
-  uint32_t rm21_sram_vmin_value;
   uint16_t gc_num_gprs;
   uint16_t gc_gsprim_buff_depth;
   uint16_t gc_parameter_cache_depth;
@@ -1251,6 +1250,8 @@ struct  atom_gfx_info_v2_4 {
   uint8_t gc_gs_table_depth;
   uint8_t gc_double_offchip_lds_buffer;
   uint8_t gc_max_scratch_slots_per_cu;
+  uint32_t sram_rm_fuses_val;
+  uint32_t sram_custom_rm_fuses_val;
 };
 
 /* 
-- 
2.7.4

